src/Symfony/Component/Validator/Constraints/UlidValidator.php

```diff
@@ -40,31 +40,54 @@ public function validate(mixed $value, Constraint $constraint): void
 
         $value = (string) $value;
 
-        if (26 !== \strlen($value)) {
+        switch ($constraint->format) {
+            case Ulid::FORMAT_BASE_32:
+                $requiredLength = 26;
+                $requiredCharset = '0123456789ABCDEFGHJKMNPQRSTVWXYZabcdefghjkmnpqrstvwxyz';
+                break;
+            case Ulid::FORMAT_BASE_58:
+                $requiredLength = 22;
+                $requiredCharset = '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz';
+                break;
+                // no default, the constraint constructor already ensures a valid format
+        }
```

Comment:
what about this?
```suggestion
        [$requiredLength, $requiredCharset] = match ($constraint->format) {
            Ulid::FORMAT_BASE_32 => [26, '0123456789ABCDEFGHJKMNPQRSTVWXYZabcdefghjkmnpqrstvwxyz'],
            Ulid::FORMAT_BASE_58 => [22, '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'],
        };
```

Reply:
I keep forgetting about this syntax, it is indeed better